### PR TITLE
feat: add js implementation of the ProteusClient

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We want validate the KMP when targeting JS to assure us that it will also be possible to integrate the library on the web.

### Solutions

- Configure JS target in the cryptography module
- Implement the ProteusClient for JS using the existing [@wireapp/cryptobox](https://www.npmjs.com/search?q=%40wireapp%2Fcryptobox) implementation 

### Testing

The common ProteusClient tests pass when running against the JS target.

### Notes (Optional)

- For simplicity is the JS ProteusClient using the in-memory engine which doesn't support any persistence.
- I've annotated many of ProteusClient methods with `suspend` since the JS cryptobox based on promises.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: cryptography/src/jsMain/kotlin/com/wire/kalium/cryptography/ProteusClient.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
+package com.wire.kalium.cryptography
+
+import com.wire.kalium.cryptography.exceptions.ProteusException
+import com.wire.kalium.cryptography.externals.Cryptobox
+import com.wire.kalium.cryptography.externals.IdentityKey
+import com.wire.kalium.cryptography.externals.MemoryEngine
+import com.wire.kalium.cryptography.externals.PreKeyBundle
+import io.ktor.util.InternalAPI
+import io.ktor.util.decodeBase64Bytes
```
Comment (typfel) [inline cryptography/src/jsMain/kotlin/com/wire/kalium/cryptography/ProteusClient.kt:9]: So this was an internal API,  is anyone aware of another Base64 implementation which works in common?

Comment (vitorhugods) [reply cryptography/src/jsMain/kotlin/com/wire/kalium/cryptography/ProteusClient.kt:9]: `ktor-utils` is available as a standalone library too. I'm not sure if maybe we could just use it on `commonMain` and use it for all platforms. Maybe creating a `Base64Encoder` for all platforms to use? 

In any case, it works!

File: cryptography/build.gradle.kts [hunk 2]
```diff
@@ -47,18 +47,26 @@ kotlin {
...
+        }
+    }
 
     sourceSets {
         val commonMain by getting {
             dependencies {
                 // coroutines
                 implementation(Dependencies.Coroutines.core)
-
+                implementation(Dependencies.Ktor.utils)
```
Comment (vitorhugods) [inline cryptography/build.gradle.kts:63]: Oh! It's added already :P

**Review by vitorhugods — APPROVED** [review]
Chills, literally chills.

**Pull request merged**